src/testcases/slashing.test.ts

```diff
@@ -1,162 +1,129 @@
-  assert.ok((await querier.statom_exchange_rate()) < 1);
+      vals[3].address,
+    );
+  });
 
-  await wait(5000);
+  test('exchange rate must be 1', async () => {
+    expect(await querier.statom_exchange_rate()).toEqual(1);
+  });
 
-  let statom_ex_rate_before_second_bond = await querier.statom_exchange_rate();
-  assert.ok(statom_ex_rate_before_second_bond < 1);
+  test('bond for statom', async () => {
+    const res = await testState.lasset.bond_for_statom(
+      testState.wallets.ownerWallet,
+      1_000,
+    );
+    expect(res.code).toEqual(0);
+    total_statom_bond_amount_before_slashing =
+      await querier.total_bond_statom_amount();
+  });
+
+  test('disconnect validator', async () => {
+    await disconnectValidator('node1');
+    await expect(testState.waitForJailed()).resolves.not.toThrow();
+    await wait(10000);
+  });
+
+  test('validate numbers', async () => {
+    expect(total_statom_bond_amount_before_slashing).toBeGreaterThan(
+      await querier.total_bond_statom_amount(),
+    );
+  });
 
-  await mustPass(
-    testState.lasset.bond_for_statom(testState.wallets.a, 2_000_000)
-  );
+  test('exchange rate must drop', async () => {
+    expect(await querier.statom_exchange_rate()).toBeLessThan(1);
+  });
 
-  let statom_ex_rate_after_second_bond = await querier.statom_exchange_rate();
+  test('bond after slashing', async () => {
+    const res = await testState.lasset.bond_for_statom(testState.wallets.a, 1);
+    expect(res.code).toEqual(0);
+    expect(await querier.statom_exchange_rate()).toBeLessThan(1);
+  });
 
-  assert.ok(statom_ex_rate_after_second_bond < 1);
+  test('validate exchange rate', async () => {
+    await wait(5000);
+    expect(await querier.statom_exchange_rate()).toBeLessThan(1);
+  });
 
-  for (let i = 0; i < 74; i++) {
-    await mustPass(
-      testState.lasset.bond_for_statom(testState.wallets.a, 2_000_000)
+  test('third bond', async () => {
+    const res = await testState.lasset.bond_for_statom(
+      testState.wallets.a,
+      2_000,
     );
-  }
-
-  await mustPass(testState.lasset.dispatch_rewards(testState.wallets.a));
-
-  // blocks 258 - 307
-  await wait(25000);
-
-  // blocks 308 - 382
-  const initial_statom_balance_a = await querier.balance_statom(
-    testState.wallets.a.key.accAddress
-  );
-  const initial_uluna_balance_a = Number(
-    (
-      await testState.wallets.a.lcd.bank.balance(
-        testState.wallets.a.key.accAddress
-      )
-    )[0].get("uluna").amount
-  );
-  for (let i = 0; i < 75; i++) {
-    await testState.lasset.send_cw20_token(
-      statomContractAddress,
+    expect(res.code).toEqual(0);
+    expect(await querier.statom_exchange_rate()).toBeLessThan(1);
+  });
+
+  test('cycle of bonds', async () => {
+    for (let i = 0; i < ITER_TIMES; i++) {
+      const res = await testState.lasset.bond_for_statom(
+        testState.wallets.a,
+        2_000,
+      );
+      expect(res.code).toEqual(0);
+    }
+  });
```

Comment:
Please add `2_000_000` timeout here. This test takes almost 700 seconds to run on my machine (default timeout is 600 seconds in this repo), so a timeout of 2000 seconds seems to be adequate even for very slow machines.

Reply:
done